src/main/charts/bitbucket/templates/shared-home-pvc.yaml

```diff
@@ -9,7 +9,14 @@ spec:
   accessModes:
     - ReadWriteMany
   {{- if .Values.volumes.sharedHome.persistentVolumeClaim.storageClassName }}
+    {{- if eq "nil" .Values.volumes.sharedHome.persistentVolumeClaim.storageClassName  }}
```

Comment:
Using "nil" as a string it a bit wird. Why not just use 

    {{- with .Values.volumes.sharedHome.persistentVolumeClaim.storageClassName  }} 
      storageClassName: {{ . | quote }}
    {{- end }}

Reply:
Will it work? What I need to have here is a literal empty string if the storage class is defined as an empty string. I think `with` will skip the stanza completely. I gave it a shot at https://play.golang.org/ and it seems like `with` works like an `if` in this case.

Reply:
Can you use `with` with `else`? This seems to work:

```
package main

import (
	"os"
	"text/template"
)

func main() {

	instanceWithValue := DataStruct{"bar"}
	instanceWithoutValue := DataStruct{}

	tmpl, err := template.New("templ").Parse(`
{{ with .Foo}}
Foo is {{ $.Foo }}
{{ else }}
Foo is empty
{{ end }}`)
	
	if err != nil {
		panic(err)
	} else {
		tmpl.Execute(os.Stdout, instanceWithValue)
		tmpl.Execute(os.Stdout, instanceWithoutValue)
	}
}

type DataStruct struct {
	Foo string
}
```

Reply:
But it's a tristate value. Empty string has a different meaning than no value and a different meaning than some value.

Reply:
I ended up doing:

`  {{- if not (kindIs "invalid" .Values.volumes.sharedHome.persistentVolumeClaim.storageClassName) }}
  storageClassName: {{ .Values.volumes.sharedHome.persistentVolumeClaim.storageClassName | quote }}
  {{- end }}`